File: totosuki/ABC/B/018.py
S = input()
N = int(input())
for _ in range(N):
  l, r = map(int, input().split())
  l, r = map(lambda x: x-1, [l,r])
  S = S[:l] + S[l:r+1][::-1] + S[r+1:]
print(S)

File: OCH_Tao/ABC/A/018.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = input.map(str => parseInt(str, 10));
const sortedN = [...N];
sortedN.sort((a, b) => b - a);
for (let i = 0; i < 3; i++) {
  console.log((sortedN.indexOf(N[i])) + 1);
}